src/main/OP.ts

```diff
@@ -225,34 +223,24 @@ function createVerifyRequestOptsFromBuilderOrExistingOpts(opts: {
   builder?: OPBuilder;
   verifyOpts?: VerifyAuthenticationRequestOpts;
 }): VerifyAuthenticationRequestOpts {
-  const subjectSyntaxTypesSupported = [];
-  if (opts.builder?.responseRegistration?.subjectSyntaxTypesSupported) {
-    if (Array.isArray(opts.builder.responseRegistration.subjectSyntaxTypesSupported)) {
-      subjectSyntaxTypesSupported.push(...opts.builder.responseRegistration.subjectSyntaxTypesSupported);
-    } else {
-      subjectSyntaxTypesSupported.push(opts.builder.responseRegistration.subjectSyntaxTypesSupported);
-    }
-  } else if (opts?.verifyOpts?.verification?.resolveOpts?.subjectSyntaxTypesSupported) {
-    if (Array.isArray(opts.verifyOpts.verification.resolveOpts.subjectSyntaxTypesSupported)) {
-      subjectSyntaxTypesSupported.push(...opts.verifyOpts.verification.resolveOpts.subjectSyntaxTypesSupported);
-    } else {
-      subjectSyntaxTypesSupported.push(opts.verifyOpts.verification.resolveOpts.subjectSyntaxTypesSupported);
-    }
+  if (opts?.builder?.resolvers.size && opts.builder?.responseRegistration) {
+    opts.builder.responseRegistration.subjectSyntaxTypesSupported = mergeAllDidMethods(
```

Comment:
yeah I struggled with it as well. I thought since our user has given us a registeration object, and also provided us with some did methods, I have to add that to the `subject_syntax_types_supported` property of the registeration object. because in the negotiation process, we want to express all the supported methods